### PR TITLE
Parse alt column structure

### PR DESCRIPTION
**Pull request by eschwartz**
Parse header row, to determine column structure

before, we were assuming that the columns were in a certain order.
However, this may not always be the case
- see http://ftp.wpc.ncep.noaa.gov/, for example,
  which is missing an `Icon` row

The parsing script is more flexible now --
 it parses the header row first, to determine the column ordering,
 and then parses each column using that same ordering

### CODE REVIEW

File: test/fixture/apache-index.html [hunk 1]
```diff
@@ -40,14 +40,14 @@
...
-    <td align="right"> 5.4M</td>
+    <td align="right"></td>
     <td>&nbsp;</td>
   </tr>
   <tr>
     <td valign="top"><img src="/icons/compressed.gif" alt="[   ]"></td>
     <td><a href="fileA.zip">fileA.zip</a></td>
     <td align="right">28-May-2009 15:09</td>
-    <td align="right"> 56K</td>
+    <td align="right"> 5.4M</td>
```
Comment (juliangruber) [inline test/fixture/apache-index.html]: why did you modify those?

**Review by juliangruber — CHANGES_REQUESTED** [review]
..the rest looks good to me!

File: test/fixture/apache-index.html (hunk 1 above)
Comment (eschwartz) [reply test/fixture/apache-index.html]: If I remember correctly, I removed the `size` field from `folderB/` (folders shouldn't have a size). But I still wanted to test that we are parsing `M` (megabyte) values correctly.

**Comment by eschwartz** [comment]
can this be merged in?

**Comment by juliangruber** [comment]
modifications to existing tests are always dangerous as they might shadow unintentional breaking changes. i'd rather we add another fixture instead. then i'll merge and release :)

**Comment by eschwartz** [comment]
I reverted the fixture changes, and also fixed some linter errors that were causing `npm test` to fail.

**Review by juliangruber — APPROVED** [review]

**Comment by juliangruber** [comment]
awesome! thank you for your work and patience :) released as 1.2.0

**Pull request merged**